sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -0,0 +1,647 @@
+/*slm-0op-1*/ simd<T, N> slm_atomic_update(simd<uint32_t, N> byte_offset,
+                                           simd_mask<N> mask = 1);
+```
+
+### atomic_update() with 1 operands (*add, *sub, *min, *max, bit_or/xor/and, store, xchg)
+```C++
+// Atomic update the USM memory locations - 1 operand (add, max, etc.).
+/*usm-1op-1*/ simd<T, N> atomic_update(T *ptr, simd<Toffset, N> byte_offset,
+                                       simd<T, N> src0, simd_mask<N> mask, props = {});
+/*usm-1op-2*/ simd<T, N> atomic_update(T *ptr, simd<Toffset, N> byte_offset,
+                                       simd<T, N> src0, props = {});
+// Similar to (usm-1op-1,2), but `byte_offset` is `simd_view`.
+/*usm-1op-3*/ simd<T, N> atomic_update(T *p, simd_view<OffsetObjT, OffsetRegionTy> byte_offset,
+                                       simd<T, N> src0, simd_mask<N> mask, props = {});
+/*usm-1op-4*/ simd<T, N> atomic_update(T *p, simd_view<OffsetObjT, OffsetRegionTy> byte_offset,
+                                       simd<T, N> src0, props = {});
+
+
+// Atomic update the memory locations referenced by device-accessor - 1 operand (add, max, etc.).
+template <atomic_op Op, typename T, int N, typename Toffset, typename AccessorT,
+          typename PropertyListT = empty_properties_t>
+/*acc-1op-1*/ simd<T, N> atomic_update(AccessorT acc, simd<Toffset, N> byte_offset,
+                                       simd<T, N> src0, simd_mask<N> mask, props = {});
+/*acc-1op-2*/ simd<T, N> atomic_update(AccessorT acc, simd<Toffset, N> byte_offset,
+                                       simd<T, N> src0, props = {});
+
+// Similar to (acc-1op-1,2), but `byte_offset` is `simd_view`.
+template <atomic_op Op, typename T, int N, typename OffsetObjT, typename AccessorT,
+          typename RegionT, typename PropertyListT = empty_properties_t>
+/*acc-1op-3*/ simd<T, N> atomic_update(AccessorT acc, simd_view<OffsetObjT, RegionT> byte_offset,
+                                       simd<T, N> src0, simd_mask<N> mask, props = {});
+/*acc-1op-4*/ simd<T, N> atomic_update(AccessorT acc, simd_view<OffsetObjT, RegionT> byte_offset,
+                                       simd<T, N> src0, props = {});
+
+// Atomic update the memory locations referenced by local-accessor (SLM) - one operand (add, max, etc.).
+template <atomic_op Op, typename T, int N, typename AccessorT>
+/*lacc-1op-1*/ simd<T, N> atomic_update(AccessorT lacc, simd<uint32_t, N> byte_offset,
+                                        simd<T, N> src0, simd_mask<1> pred = 1);
+
+// Atomic update the shared local memory (SLM) - one operand (add, max etc.).
+template <atomic_op Op, typename T, int N>
+/*slm-1op-1*/ simd<T, N> slm_atomic_update(simd<uint32_t, N> byte_offset,
+                                           simd<T, N> src0, simd_mask<N> mask = 1);
+```
+### atomic_update() with 2 operands (cmpxchg, fcmpxchg)
+```C++
+// Atomic update the USM memory locations - 2 operand: *cmpxchg.
```

Comment:
Sorry what is the `*` here for?

Reply:
I replaced `*cmpxchg` with `cmpxchg, fcmpxchg` in 161e753c6100f9bd74ec32fde32237c6957aa4c9 and 0be57b6470981fb5078dbb17241f04bd6e3c8c55